components/DataGenerator.js

```diff
@@ -0,0 +1,220 @@
+import { uid } from "uid";
+import Chance from "chance";
+import styled from "styled-components";
+import { exampleData } from "@/lib/db";
+import {
+  StyledButton,
+  StyledFlexColumnWrapper,
+} from "@/SharedStyledComponents";
+
+const StyledFlexWrapper = styled(StyledFlexColumnWrapper)`
+  flex-direction: row;
+  gap: 1rem;
+  margin: 1rem;
+`;
+
+const StyledGeneratorTitle = styled.h1`
+  font-weight: 600;
+`;
+
+const StyledDevButton = styled(StyledButton)`
+  width: fit-content;
+  margin: 0;
+  padding: 0.3rem;
+`;
+
+const StyledRedButton = styled(StyledDevButton)`
+  background-color: red;
+`;
+
+const StyledSubTitle = styled.h2`
+  padding: 0.5rem;
+  margin: 0.5rem;
+  font-size: 1.3rem;
+  font-weight: 600;
+`;
+
+const chance = new Chance();
+
+const dateOptions = {
+  weekday: "long",
+  year: "numeric",
+  month: "long",
+  day: "numeric",
+};
+
+// generating random emotion entries for a day, takes endhour and daysTimestamp as paramaters;
+// math model: sin
+function generateDaysEntries(endHour, daysTimestamp) {
+  const fullDate = new Date(daysTimestamp);
+
+  const date = new Intl.DateTimeFormat("en-US", dateOptions).format(fullDate);
+
+  //variables for sinus curve
+  const a = chance.integer({ min: -50, max: 50 });
+  const b = chance.floating({ min: -3, max: 3 });
+  const c = chance.floating({ min: 4, max: 8 });
+  const d = chance.integer({ min: 30, max: 60 });
+  //building array for sensible hours
+  const basicArray = [...Array(endHour + 1).keys()];
+  const hoursToFill = basicArray.slice(5);
+
+  const hourlyEntries = hoursToFill.map((hour) => {
+    const minutes = chance.integer({ min: 0, max: 59 });
+    const time = hour + ":" + minutes.toString().padStart(2, "0");
+
+    const object = {
+      id: uid(),
+      date,
+      time,
+      timeAndDate: date + ", " + time,
```

Comment:
Same as above - but no have to:
```suggestion
      timeAndDate: `${date}, ${time}`,
```